sound/soc/sof/pcm.c

```diff
@@ -528,7 +528,7 @@ static int sof_pcm_new(struct snd_soc_pcm_runtime *rtd)
 	struct snd_sof_pcm *spcm;
 	struct snd_pcm *pcm = rtd->pcm;
 	struct snd_soc_tplg_stream_caps *caps;
-	int ret = 0, stream = SNDRV_PCM_STREAM_PLAYBACK;
+	int ret, stream = SNDRV_PCM_STREAM_PLAYBACK;
```

Comment:
wrong as is, but please remove "ret" completely in this function and convert below to "return 0"